.github/workflows/pull_request_soundness.yml

```diff
@@ -50,64 +38,10 @@ on:
 
 ## We are cancelling previously triggered workflow runs
 concurrency:
-  group: ${{ github.workflow }}-${{ github.ref }}
+  group: ${{ github.workflow }}-${{ github.ref }}-soundness
```

Comment:
I ... don't think this addition is needed? Did you run into a problem and this solved it?